sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/implementation/ReactorSessionCacheTest.java

```diff
@@ -0,0 +1,273 @@
+            .verifyComplete();
+    }
+
+    @Test
+    void shouldExplicitEvictRemoveSession() {
+        final ReactorSessionCache cache = createSessionCache();
+        final Connection connection = mock(Connection.class);
+
+        final String session0Name = "session-0";
+        final Sinks.Many<AmqpEndpointState> session0aEndpointStates = endpointStatesSink();
+        final ReactorSession session0a = session(session0Name, session0aEndpointStates, null);
+        final ReactorSession session0b = session(session0Name, endpointStatesSink(), null);
+
+        final HashMap<String, Deque<ReactorSession>> cacheLoaderLookup = new HashMap<>();
+        cacheLoaderLookup.put(session0Name, sessions(session0a, session0b));
+        final CacheLoader cacheLoader = new CacheLoader(cacheLoaderLookup);
+
+        StepVerifier.create(cache.getOrLoad(Mono.just(connection), session0Name, cacheLoader))
+            .expectNext(session0a)
+            .verifyComplete();
+
+        // explicitly evict session0a.
+        Assertions.assertTrue(cache.evict(session0Name));
+
+        // Since the session0a was evicted, the cache should return the new session, session0b.
+        StepVerifier.create(cache.getOrLoad(Mono.just(connection), session0Name, cacheLoader))
+            .expectNext(session0b)
+            .verifyComplete();
+    }
+
+    @Test
+    void shouldNotEvictSessionIfOwnerDisposed() {
+        final ReactorSessionCache cache = createSessionCache();
+        final Connection connection = mock(Connection.class);
+
+        final String session0Name = "session-0";
+        final Sinks.Many<AmqpEndpointState> session0EndpointStates = endpointStatesSink();
+        final ReactorSession session0 = session(session0Name, endpointStatesSink(), null);
+
+        final HashMap<String, Deque<ReactorSession>> cacheLoaderLookup = new HashMap<>();
+        cacheLoaderLookup.put(session0Name, sessions(session0));
+        final CacheLoader cacheLoader = new CacheLoader(cacheLoaderLookup);
+
+        StepVerifier.create(cache.getOrLoad(Mono.just(connection), session0Name, cacheLoader))
+            .expectNext(session0)
+            .verifyComplete();
+
+        // Signal that the cache owner (Connection) is disposed.
+        cache.setOwnerDisposed();
+        // Signal that session0a is terminated after owner disposal.
+        session0EndpointStates.emitComplete(Sinks.EmitFailureHandler.FAIL_FAST);
+
+        // Since the owner is disposed, the cache should not evict the session0 even if it's terminated.
+        StepVerifier.create(cache.getOrLoad(Mono.just(connection), session0Name, cacheLoader))
+            .expectNext(session0)
+            .verifyComplete();
+    }
+
+    private ReactorSessionCache createSessionCache() {
+        return new ReactorSessionCache(CONNECTION_ID, NAMESPACE, handlerProvider, reactorProvider, OPEN_TIMEOUT,
+            logger);
+    }
+
+    private Sinks.Many<AmqpEndpointState> endpointStatesSink() {
```

Comment:
nit: i was curious why this is not also static?

Reply:
ah, yes, this can be _static_ and can be aligned with other similar _static_ methods